packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -1977,6 +1985,59 @@ export class ContainerRuntime
 		this.removeAllListeners();
 	}
 
+	/**
+	 * {@inheritDoc @fluidframework/runtime-definitions#IContainerRuntimeBase.isSnapshotFetchRequired}
+	 */
+	public isSnapshotFetchRequired(pathParts: string[]): boolean {
+		validatePrecondition(
+			this.snapshotWithContents !== undefined,
+			"Use this api in case we fetched the complete snapshot along with the blob contents",
+		);
+		// Find the snapshotTree inside the snapshot based on the path as given in the request.
+		const hasIsolatedChannels = rootHasIsolatedChannels(this.metadata);
+		const snapshotTreeForPath = this.getSnapshotTreeForPath(
```

Comment:
quickly glancing at getSnapshotTreeForPath(), I think it's wrong for general case. We can't add ".channels" on every level. Container Runtime should add it only once.
If we ever implement it for sub-data stores or DDSs, then appropriate node in a call tree will add another one as needed, i.e. if it comes to data store context, and it knows it has ".channels" underneath it, then it will add it there.

Reply:
Removed this logic as it not required anymore.